Automattic/wp-calypso
# DevDocs: Fix UI/App Components Search Filter Summary

## Background
The `devdocs/design` UI and App components pages leverage common Search\Filter capabilities which is supposed to provide a Filter Summary functionality that does not work to day. This PR aims at fixing the Filter summary functionality and also does some clean-up along the way...

- We move the Search\Filter capability into its own file `search-collection.jsx` to make it a reusable class `Collection`.
- We then import this Collection class into the UI and App component related files and make use of it to list the collection of examples UI/App components.
- Lastly, and most importantly, we fix the Filter Summary feature which wasn't working previously.
- Since we're at it, we also ES6'ify the code along the way and sort the list of components for a better user experience while filtering (representative of a dictionary/phonebook)

## Testing

- Launch [localhost:devdocs/design](http://calypso.localhost:3000/devdocs/design) and make sure you see all the UI Components. For comparison you can use [wpcalypso:devdocs/design](https://wpcalypso.wordpress.com/devdocs/design)
- Launch [localhost:devdocs/app-components](http://calypso.localhost:3000/devdocs/app-components) and make sure you see all the Application Components. For comparison you can use [wpcalypso:devdocs/app-components](https://wpcalypso.wordpress.com/devdocs/app-components)
- Ensure you see the components in alphabetical sorted order.
- Now for the main Filter Summary test, enter a filter keyword like (say) `gr`, you should see the result as depicted below:

![image](https://cloud.githubusercontent.com/assets/14303849/13558394/18ffaff4-e3d2-11e5-9bec-b803096ac824.png)

## Reviewers
@aduth @nb @mtias @apeatling 

## Issues
Closes #3811
